### PR TITLE
feat(datadog_events sink): Use log namespacing and semantic meaning

### PR DESCRIPTION
**Pull request by spencergilbert**
Closes: #16484

### CODE REVIEW

File: src/sinks/datadog/events/sink.rs [hunk 1]
```diff
@@ -55,34 +55,28 @@ async fn ensure_required_fields(event: Event) -> Option<Event> {
...
+            .message_path()
+            .expect("message is required (make sure the \"message\" semantic meaning is set)");
+        log.rename_key(message_path.as_str(), event_path!("text"))
     }
 
     if !log.contains("host") {
-        if let Some(host) = log.remove(log_schema.host_key()) {
-            log.insert("host", host);
+        if let Some(host_path) = log.host_path() {
+            log.rename_key(host_path.as_str(), event_path!("host"));
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply src/sinks/datadog/events/sink.rs:67]: It's also not necessarily between the metadata and the event, which makes the naming challenging

File: src/sinks/datadog/events/config.rs [hunk 2]
```diff
@@ -166,7 +168,13 @@ impl SinkConfig for DatadogEventsConfig {
     }
 
     fn input(&self) -> Input {
-        Input::log()
+        let requirement = schema::Requirement::empty()
+            .required_meaning("message", Kind::bytes())
+            .optional_meaning("host", Kind::bytes())
+            .optional_meaning("timestamp", Kind::timestamp())
+            .optional_meaning("source_type", Kind::bytes());
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply src/sinks/datadog/events/config.rs]: Fixed in [2383b96](https://github.com/vectordotdev/vector/pull/16499/commits/2383b966a0802161f6d54df7046973a9f17f0feb)

**Review by fuchsnj — APPROVED** [review]

File: src/sinks/datadog/events/sink.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (fuchsnj) [reply src/sinks/datadog/events/sink.rs:67]: maybe just `move_field`?

[the remark this answers is not in the document]
Comment (spencergilbert) [reply src/sinks/datadog/events/sink.rs:67]: 👍 I'd back that change. I'd love to get consistent in the `key` vs `field` phrasing, I think we flipflop between them.

**Pull request merged**